### PR TITLE
[DT-714 and DT-715] Add Redirect and 3DS

### PR DESCRIPTION
**Pull request by ricardo-frango**
**What does this PR do?**

Added redirect and 3DS (not handling the result yet)

**Database changed?**

   No

**Where should the reviewer start?**

- start by testing the flow of a normal card, than a card with redirect

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [DT-715](https://aptoide.atlassian.net/browse/DT-715)
  Flow on how to test this or QA Tickets related to this use-case: [DT-714](https://aptoide.atlassian.net/browse/DT-714)

**What are the relevant tickets?**

  Tickets related to this pull-request: [DT-715](https://aptoide.atlassian.net/browse/DT-715)
  Tickets related to this pull-request: [DT-714](https://aptoide.atlassian.net/browse/DT-714)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[DT-715]: https://aptoide.atlassian.net/browse/DT-715?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DT-714]: https://aptoide.atlassian.net/browse/DT-714?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DT-715]: https://aptoide.atlassian.net/browse/DT-715?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DT-714]: https://aptoide.atlassian.net/browse/DT-714?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by G0retZ — CHANGES_REQUESTED** [review]

File: aptoide-network/src/main/java/cm/aptoide/pt/aptoide_network/di/NetworkModule.kt [hunk 1]
```diff
@@ -53,7 +53,7 @@ object NetworkModule {
     versionCodeInterceptor: VersionCodeInterceptor,
   ): OkHttpClient {
     val interceptor = HttpLoggingInterceptor()
-    interceptor.level = HttpLoggingInterceptor.Level.BASIC
+    interceptor.level = HttpLoggingInterceptor.Level.BODY
```
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply aptoide-network/src/main/java/cm/aptoide/pt/aptoide_network/di/NetworkModule.kt]: fixed

File: aptoide-network/src/main/java/cm/aptoide/pt/aptoide_network/di/NetworkModule.kt [hunk 2]
```diff
@@ -33,7 +33,7 @@ object NetworkModule {
     languageInterceptor: LanguageInterceptor,
   ): OkHttpClient {
     val interceptor = HttpLoggingInterceptor()
-    interceptor.level = HttpLoggingInterceptor.Level.BASIC
+    interceptor.level = HttpLoggingInterceptor.Level.BODY
```
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply aptoide-network/src/main/java/cm/aptoide/pt/aptoide_network/di/NetworkModule.kt]: fixed

File: payments/base/payment-manager/src/main/java/cm/aptoide/pt/payment_manager/payment/PaymentMethod.kt [hunk 3]
```diff
@@ -57,7 +57,9 @@ val emptyPaymentMethod = object : PaymentMethod<String> {
     skills = false
   )
 
-  override suspend fun createTransaction(paymentDetails: String): Transaction {
-    TODO("Don't need to be implemented")
+  override suspend fun createTransaction(
```
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply payments/base/payment-manager/src/main/java/cm/aptoide/pt/payment_manager/payment/PaymentMethod.kt]: fixed

File: payments/osp-handler/src/main/java/cm/aptoide/pt/osp_handler/PaymentActivity.kt [hunk 4]
```diff
@@ -28,4 +29,16 @@ class PaymentActivity : AppCompatActivity() {
...
   }
+
+  override fun onNewIntent(intent: Intent?) {
+    super.onNewIntent(intent)
+    handleIntent(intent)
+  }
+
+  private fun handleIntent(intent: Intent?) {
+    val data = intent?.data
+    if (data != null && data.toString().startsWith("adyencheckout")) {
```
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply payments/osp-handler/src/main/java/cm/aptoide/pt/osp_handler/PaymentActivity.kt]: fixed

File: payments/osp-handler/src/main/java/cm/aptoide/pt/osp_handler/PaymentScreenContentProvider.kt [hunk 5]
```diff
@@ -1,8 +1,12 @@
 package cm.aptoide.pt.osp_handler
 
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
 import androidx.compose.runtime.Composable
 import cm.aptoide.pt.payment_manager.manager.PurchaseRequest
 
 interface PaymentScreenContentProvider {
+  fun handleIntent(context: AppCompatActivity, intent: Intent, )
```
[the remark this answers is not in the document]
Comment (ricardo-frango) [reply payments/osp-handler/src/main/java/cm/aptoide/pt/osp_handler/PaymentScreenContentProvider.kt]: fixed

**Review by G0retZ — APPROVED** [review]

**Pull request merged**